### PR TITLE
Add support for wildcard origins (fixes #14)

### LINKED ISSUE #14: Multiple origins with wildcards
Hi guys, I read all the code but is not clear to me. This middleware supports multiple origins with wildcards.

Example: `[ 'https://*.domain.com', 'https://*.domainmagic.com' ]`

Best regards.

### PR DESCRIPTION
**Pull request by mikhailidi**
This PR is related to #14 and allows using wildcard configuration for allowed origins.

Based on your comment https://github.com/tuupola/cors-middleware/issues/14#issuecomment-607736091, I decided to add Settings class which extends original `Neomerx\Cors\Strategies\Settings` and adds some extra checks as proposed multiple times in #14.

### CODE REVIEW

**Comment by mikhailidi** [comment]
Hey hey @tuupola 
Do you have time to review it?

**Comment by mikhailidi** [comment]
@tuupola ping :)

File: tests/SettingsTest.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        $this->testObject = new Settings();
+    }
+
+    /**
+     * @dataProvider wildcardOriginDataProvider
+     */
+    public function testIsRequestOriginAllowed(string $origin, array $allowedOrigins, bool $expected): void
+    {
+        $requestOriginMock = $this->createMock(ParsedUrl::class);
+        $requestOriginMock->method('getOrigin')
```
Comment (tuupola) [inline tests/SettingsTest.php:29]: Should use " instead of ' to match other code.

**Comment by tuupola** [comment]
LGTM. Some small formatting issues. Should use " instead of ' everywhere. No any other reason than that is what is used elsewhere in the code.  I was pondering between `fnmatch()` and `preg_match()` but I think  `fnmatch()` goes well with the default wildcard of `*`.

**Comment by tuupola** [comment]
I can update the README. There are some other chores I need to do too.

**Comment by tuupola** [comment]
Released as 1.3.0: Thanks!

**Pull request merged**